pac_man/pacman.py

```diff
@@ -0,0 +1,141 @@
+from matplotlib import pyplot as plt
+import numpy as np
+from random import randint
+
+
+class OutsideOfLegalPath(Exception):
+    pass
+
+
+class Agent:
+    def __init__(self, current_node=None):
+        self.current_node = current_node
+
+    def move(self):
+        children_nodes = self.current_node.children_nodes
+        self.current_node = children_nodes[randint(0, len(children_nodes) - 1)]
+
+
+class Ghost(Agent):
+    def __init__(self, current_node, killable=False):
+        super().__init__(current_node)
+        self.killable = killable
+
+
+class PacMan(Agent):
+    def __init_(self, current_node):
+        super().__init__(heigt)
+
+
+class Node:
+    def __init__(self, position, children_nodes=None, reward=1):
+        self.position = position
+        self.reward = reward
+        if (children_nodes is None):
+            self.children_nodes = []
+        else:
+            self.children_nodes = children_nodes
+
+    def __str__(self):
+        return ' position : {pos} \n children nb : {children_nb} \n'.\
+            format(children_nb=str(len(self.children_nodes)),
+                   pos=self.position)
+
+
+class Board:
+    def __init__(self, board_file_name):
+        self.agents = []
+        board_nodes, board_outline = Board.construct_board(board_file_name)
+        self.board_nodes = board_nodes
+        self.board_outline = board_outline
+
+    @staticmethod

```

Comment:
You can check for those unwanted characters before commiting your changes with `git diff --cached` if I am not mistaken.